src/compiler/checker.ts

```diff
@@ -12075,6 +12076,12 @@ namespace ts {
             return !!(getObjectFlags(type) & ObjectFlags.Reference) && (<TypeReference>type).target === globalArrayType;
         }
 
+        function isPromiseLikeType(type: Type): boolean {
+            const globalPromiseType = getGlobalPromiseType(/*reportErrors*/ false); // this is only called from the language service, so don't report errors if the promise type doesn't exist
+            return getObjectFlags(type) & ObjectFlags.Reference && (<TypeReference>type).target === globalPromiseType ||

```

Comment:
i would instead call `return !!getPromisedTypeOfPromise(type);` it does a few other things than looking up the global Promise type reference.